### DISCUSSION TITLE
use of mainClass

### DISCUSSION THREAD

**Discussion by bmarwell** [jreleaser/jreleaser#134] [Q&A] [answered]
Hi,

I am working on a multi-module project where actually two artifacts can drop out of the build.
For now, it is a runnable binary using a meecrowave server. But in the future, other servers might be added.

So I wonder: what happens to the "java.mainClass" key? I might have multiple of them in the future, but I can currently control those via the `distributions` configuration key. And at the moment, I have no `mainClass` configured.

**Comment by aalmiray** [comment, marked as answer]
The value of `java.mainClass` is currently used by the Jbang packager and the Jlink assembler (for non-modular applications) to generate launcher scripts. It can also be used on any template (such as the e-mail or discussions templates).

If you're not using the Jbang packager, or the Jlink assembler they you may leave this value out.

If you need more than one value then you can set it at each `distribution`, with the value set at `project` being the default one.

**Reply by bmarwell** [reply]
Thanks! Got it!

**Reply by javydreamercsw** [reply]
Seems like it can be marked as answered.

**Reply by bmarwell** [reply]
> Seems like it can be marked as answered.

Thanks, didn't know this works just like SO :)
